app/templating/template_context.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from app.templating.metadata_context import build_metadata
+from app.templating.summary.section import Section
+from app.utilities.date_utils import to_date
+
+
+from flask import g
+
+logger = logging.getLogger(__name__)
+
+
+def build_questionnaire_rendering_context(schema_json, state, metadata):
+    """
+    Build questionnaire context containing metadata and content state of the questionnaire
+    :param schema_json: schema of the questionnaire
+    :param state: state of the questionnaire
+    :param metadata: user metadata
+    :return: questionnnaire context
+    """
+    meta = build_metadata(schema_json, metadata)
+    return _build_rendering_context(meta, state)
+
+
+def build_summary_rendering_context(schema_json, answers, metadata):
+    """
+    Build questionnaire summary context containing metadata and content from the answers of the questionnaire
+    :param schema_json: schema of the current questionnaire
+    :param answers: all of the answers to the questionnaire
+    :param metadata: user metadata
+    :return: questionnaire summary context
+    """
+    meta = build_metadata(schema_json, metadata)
+    content = _build_summary(schema_json, answers)
+    return _build_rendering_context(meta, content)
+
+
+def get_schema_template_context(metadata, aliases, answers):
+    """
+    Build questionnaire schema context containing exercise and answers
+    :param metadata: user metadata
+    :param aliases: alias mapping of friendly names to answer ids
+    :param answers: all the answers for the given questionnaire
+    :return: questionnaire schema context
+    """
+    return {
+      "exercise": _build_exercise(metadata),
+      "answers": _map_alias_to_answers(aliases, answers),
+    }
+
+
+def _build_rendering_context(meta, content):
+    return {
+      "meta": meta,
```

Comment:
I don't think we should be returning meta and content from this method, I think we should just return the content. It would make testing this class a lot simpler (no need to mock metadata) and the metadata is always treated separately to the content (we explicitly pull it out of this dict to pass it to the template).